### ISSUE TITLE
GraphQL - Exception when trying to use a token that is not assigned to a schema

### ISSUE THREAD

**Issue by OscarBarrett** [craftcms/cms#6043]
### Description
If you attempt to use a GraphQL token that is not assigned to a schema, GraphQL responds with an exception.

Instead it should provide a clearer error message, or you should be prevented from creating a token when there are no schemas to assign it to. If the latter, then there would need to be consideration around the deletion of schemas that are associated with active tokens.

```json
{
  "name": "Exception",
  "message": "Return value of craft\\controllers\\GraphqlController::_schema() must be an instance of craft\\models\\GqlSchema, null returned",
  "code": 0,
  "type": "TypeError",
  "file": "/app/vendor/craftcms/cms/src/controllers/GraphqlController.php",
  "line": 190,
  "stack-trace": [
    "#0 /app/vendor/craftcms/cms/src/controllers/GraphqlController.php(92): craft\\controllers\\GraphqlController->_schema(Object(craft\\services\\Gql))",
    "#1 [internal function]: craft\\controllers\\GraphqlController->actionApi()",
    "#2 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)",
    "#3 /app/vendor/yiisoft/yii2/base/Controller.php(157): yii\\base\\InlineAction->runWithParams(Array)",
    "#4 /app/vendor/craftcms/cms/src/web/Controller.php(178): yii\\base\\Controller->runAction('api', Array)",
    "#5 /app/vendor/yiisoft/yii2/base/Module.php(528): craft\\web\\Controller->runAction('api', Array)",
    "#6 /app/vendor/craftcms/cms/src/web/Application.php(291): yii\\base\\Module->runAction('graphql/api', Array)",
    "#7 /app/vendor/yiisoft/yii2/web/Application.php(103): craft\\web\\Application->runAction('graphql/api', Array)",
    "#8 /app/vendor/craftcms/cms/src/web/Application.php(276): yii\\web\\Application->handleRequest(Object(craft\\web\\Request))",
    "#9 /app/vendor/yiisoft/yii2/base/Application.php(386): craft\\web\\Application->handleRequest(Object(craft\\web\\Request))",
    "#10 /app/web/index.php(21): yii\\base\\Application->run()", "#11 {main}"
  ]
}
```

### Steps to reproduce

1. Create a token before creating a schema.
2. Make a GraphQL request using the token

### Additional info

- Craft version: 3.4.18
- PHP version: 7.3
- Database driver & version: MySQL 5.7
- Plugins & versions: N/A

**Comment by brandonkelly** [comment]
Thanks for reporting that! Just fixed for the next release.

Going forward you will get a 400 response with the message, “Invalid Authorization header”. 

> or you should be prevented from creating a token when there are no schemas to assign it to. If the latter, then there would need to be consideration around the deletion of schemas that are associated with active tokens.

It’s not possible to create a token without a schema; the way you would get into this situation is if you deleted the schema after already creating the token for it. Which is by design – I don’t think Craft should auto-delete tokens when a schema is deleted; you may wish to reassign the token to a different schema, without having to update your application.

**Comment by OscarBarrett** [comment]
Thanks for the quick fix.

RE creating a token without a schema: I ran into this while working on upgrading an app from Craft 2 to 3, after resetting my database with an existing `project.yaml` file that included both a public and private schema. During the upgrade process a plugin migration failed on the first run but passed on the subsequent run, and no private schema was shown in the UI. I was able to create a token with only a public schema available (which can't be assigned to a token).

e.g. https://d.pr/v/QnIYyZ
